galadriel/tools/web3/onchain/solana/spl_token.py

```diff
@@ -90,11 +91,105 @@ def forward(self, user_address: str, token_address: str) -> Optional[float]:
             return None
 
 
+class TransferTokenTool(SolanaBaseTool):
+    """Tool for transferring SPL tokens between user
+
+    Attributes:
+        name (str): Tool identifier
+        description (str): Description of the tool's functionality
+        inputs (dict): Schema for required input parameters
+        output_type (str): Type of data returned by the tool
+    """
+
+    name = "transfer_token"
+    description = "Transfers SPL tokens between users."
+    inputs = {
+        "recipient_address": {
+            "type": "string",
+            "description": "The address of the recipient.",
+        },
+        "token_address": {
+            "type": "string",
+            "description": "The SPL token mint address.",
+        },
+        "amount": {
+            "type": "number",
+            "description": "The amount of tokens to transfer.",
+        },
+    }
+    output_type = "string"
+
+    def __init__(self, wallet: SolanaWallet, *args, **kwargs):
+        super().__init__(wallet, *args, **kwargs)
+
+    def forward(self, recipient_address: str, token_address: str, amount: float) -> str:
+        """Transfer SPL tokens between user addresses.
+
+        Args:
+            recipient_address (str): The recipient's Solana wallet address
+            token_address (str): The token's mint address
+            amount (float): The amount of tokens to transfer
+
+        Returns:
+            str: The transaction signature if successful, error message if failed
+        """
+        try:
+            keypair = self.wallet.get_wallet()
+            sender_pubkey = Pubkey.from_string(self.wallet.get_address())
+            recipient_pubkey = Pubkey.from_string(recipient_address)
+            token_pubkey = Pubkey.from_string(token_address)
+
+            # Initialize SPL token client
+            spl_client = Token(self.client, token_pubkey, TOKEN_PROGRAM_ID, keypair)  # type: ignore
```

Comment:
Consider named arguments when passing a few arguments of the same type (here `PublicKey`)